src/protocol/socket.ts

```diff
@@ -133,7 +133,9 @@ function onSocketClose() {
   };
 }
 
-function onSocketError() {
+function onSocketError(evt: Event) {
+  console.error("Socket Error", evt);
+  willClose = true;
```

Comment:
Looks good! Can you leave a comment so it's obvious why we have to set `willClose = true` twice?

Reply:
Yes!